test/unit/spec/common/errorDisplay-Spec.js

```diff
@@ -0,0 +1,232 @@
+                    error: {
+                        name: 'Oh no',
+                        message: 'Disaster',
+                        error: 'calamity\ndisaster\ndespair',
+                    }
+                }
+            }
+        }
+    });
+    let cssBaseClass;
+
+    describe('The error display module', () => {
+        it('loads', () => {
+            expect(ErrorDisplay).not.toBe(null);
+        });
+
+        it('has expected functions', () => {
+            const functions = ['make', 'defaultAdvice', 'cssBaseClass'];
+            functions.forEach((f) => {
+                expect(ErrorDisplay[f]).toBeDefined();
+            });
+        });
+    });
+
+    describe('The UI module attribute defaultAdvice', () => {
+        it('should return a string about contacting support', () => {
+            expect(ErrorDisplay.defaultAdvice).toContain('kbase.us/support');
+        });
+    });
+
+    describe('The UI module attribute cssBaseClass', () => {
+        it('should return a string to use as a CSS base class', () => {
+            expect(ErrorDisplay.cssBaseClass).toContain('kb-error-display');
+        });
+    });
+
+    describe('An errorDisplay instance', () => {
+        let errorDisplayInstance;
+        beforeEach(() => {
+            errorDisplayInstance = ErrorDisplay.make({
+                model: validInput,
+            });
+        });
+
+        it('has a make function that returns an object', () => {
+            expect(errorDisplayInstance).not.toBe(null);
+        });
+
+        it('has the required methods', () => {
+            expect(errorDisplayInstance.start).toBeDefined();
+            expect(errorDisplayInstance.stop).toBeDefined();
+        });
+
+        it('should start and render an error', async () => {
+            const node = document.createElement('div');
+            cssBaseClass = ErrorDisplay.cssBaseClass;
+
+            await errorDisplayInstance.start({ node: node });
+            expect(node.classList).toContain(`${cssBaseClass}__container`);
+            expect(node.innerHTML).toContain('Error stacktrace');
+        });
+
+        it('should clean up after itself', async() => {
+            const node = document.createElement('div');
+            cssBaseClass = ErrorDisplay.cssBaseClass;
+
+            await errorDisplayInstance.start({ node: node });
+            expect(node.classList).toContain(`${cssBaseClass}__container`);
+            await errorDisplayInstance.stop();
+            expect(node.innerHTML).toBe('');
+        });
+    });
+
+    describe('The ErrorDisplay module should render different errors', () => {
+
+        const invalidArgs = [
+            null,
+            undefined,
+            {},
+            {this: 'that'},
+            Props.make({}),
+        ]
+        invalidArgs.forEach((input) => {
+            it(`should throw an error with invalid input ${ JSON.stringify(input) }`, () => {
+                try {
+                    ErrorDisplay.make(input)
```

Comment:
You might want to include something here that'll catch if it doesn't throw an error. Maybe use the `done.fail` Jasmine idiom?

Reply:
You could also use `expect(() => ErrorDisplay.make(input)).toThrow()` (or whatever the syntax is, I forget off hand) instead. That might be simpler.

Reply:
For some reason, when I used that syntax, the tests would not pass -- I will reinvestigate as I have since gained more insight into possible reasons why jasmine error tests might erroneously fail.

Reply:
OK, just reinvestigated this. I tried a couple of different things:

```js
        const invalidArgs = [null, undefined, {}, { this: 'that' }, Props.make({})];
        invalidArgs.forEach((input) => {
            it(`should throw an error with invalid input ${JSON.stringify(input)}`, () => {
                try {
                    ErrorDisplay.make(input);
                    fail(`Error not thrown with invalid input ${JSON.stringify(input)}`);
                } catch (error) {
                    expect(error.toString()).toContain('Invalid input for the ErrorDisplay module');
                    expect(error).toBeInstanceOf(Error);
                }
            });
            it(`should pass a toThrow() test with invalid input ${JSON.stringify(input)}`, () => {
                expect(ErrorDisplay.make(input)).toThrow();
            });
            it(`should pass a toThrow(Error, /regex/) test with invalid input ${JSON.stringify(
                input
            )}`, () => {
                expect(ErrorDisplay.make(input)).toThrow(
                    Error,
                    /Invalid input for the ErrorDisplay module/
                );
            });
```

output:
<img width="749" alt="Screenshot 2021-02-01 at 15 49 57" src="https://user-images.githubusercontent.com/556057/106532516-2cdd1580-64a5-11eb-9f78-1c1493b0ac90.png">

```js
    ✖ should pass a toThrow(Error, /regex/) test with invalid input {"this":"that"}
      Chrome Headless 88.0.4324.96 (Mac OS 11.1.0)
    Error: Invalid input for the ErrorDisplay module: {"this":"that"}
        at factory (http://localhost:9876/narrative/static/kjs/common/errorDisplay.js:6:415)
        at Object.make (http://localhost:9876/narrative/static/kjs/common/errorDisplay.js:7:462)
        at UserContext.<anonymous> (test/unit/spec/common/errorDisplay-Spec.js:100:37)
        at <Jasmine>

    ✖ should pass a toThrow() test with invalid input {"this":"that"}
      Chrome Headless 88.0.4324.96 (Mac OS 11.1.0)
    Error: Invalid input for the ErrorDisplay module: {"this":"that"}
        at factory (http://localhost:9876/narrative/static/kjs/common/errorDisplay.js:6:415)
        at Object.make (http://localhost:9876/narrative/static/kjs/common/errorDisplay.js:7:462)
        at UserContext.<anonymous> (test/unit/spec/common/errorDisplay-Spec.js:95:37)
        at <Jasmine>
```

Reply:
Good enough for me!